File: practice_questions/leetcode/plus-one.py
# Given an array of numbers, return an array that reprents the digit plus one of all
# the numbers in the input array:

# Input = [1,2,3]
# Output = [1,2,4] (input is equal to 123, which plus one is 124)

def plusOne(digits):
    answer = []
    strNums = ""
    for num in digits:
        strNums += str(num)
    
    intNums = int(strNums) + 1

    for num in str(intNums):
        answer.append(int(num))
    
    return answer

File: cpp/practice_questions/plus_one.cpp
// For a given vector, return a vector that represents it in digits, +1;

// Input => [1,2,3]
// Output => [1,2,4]

// Input => [9,9,9]
// Output => [1,0,0,0]

vector<int> plusOne(vector<int> digits)
{
    for (int i = digits.size() - 1; i >= 0; i--)
    {
        if (digits[i] == 9)
        {
            digits[i] = 0;
        }
        else
        {
            digits[i]++;
            return digits;
        }

        digits[0] = 1;
        digits.push_back(0);

        return digits;
    }
}